src/hotspot/share/cds/runTimeClassInfo.hpp

```diff
@@ -173,12 +175,12 @@ class RunTimeClassInfo {
     return (char*)(address(this) + verifier_constraint_flags_offset());
   }
 
-  InstanceKlass** nest_host_addr() {
-    assert(_klass->is_hidden(), "sanity");
-    return (InstanceKlass**)(address(this) + nest_host_offset());
-  }
   InstanceKlass* nest_host() {
-    return *nest_host_addr();
+    if (ArchiveBuilder::is_active() && ArchiveBuilder::current()->is_in_buffer_space((address)this)) {
+      return ArchiveBuilder::current()->offset_to_buffered<InstanceKlass*>(_nest_host_offset);
```

Comment:
I think this function is never called under this condition. Maybe the `if` should be turned into an assert:

```
assert(!ArchiveBuilder::is_active(), "not called when dumping archive");
```